core/recourse_iterator.py

```diff
@@ -46,19 +45,21 @@ def iterate_k_recourse_paths(
             A sequence of DataFrames where each DataFrame is a path and each
             row of a given DataFrame is a single step in the path.
         """
+        paths = []
+
+        # Generate the instructions for how to start the paths.
         all_instructions = self.recourse_method.get_all_recourse_instructions(
             poi
         )
-        # Start the paths from the POI
-        counterfactuals = []
-        for instructions in all_instructions:
+
+        # Follow the instructions and continue each path
+        for direction_index, instructions in enumerate(all_instructions):
+            if instructions is None:
+                paths.append(poi.to_frame().T)
```

Comment:
Is it clear in the results that recourse was not successful?

Reply:
I think so. Failed recourse shows up as "stub paths" where the only point in the path is the POI. So you can check for failed recourse by filtering on path length.